bridge-history-api/internal/logic/l1_fetcher.go

```diff
@@ -45,6 +45,9 @@ type L1FetcherLogic struct {
 	crossMessageOrm *orm.CrossMessage
 	batchEventOrm   *orm.BatchEvent
 
+	lastFinalizedBatchIndex            *uint64
+	nextPendingLastFinalizedBatchIndex uint64
```

Comment:
is this design like pre-commit, then commit? could these two variables be merged into one?

Reply:
`lastFinalizedBatchIndex` should be updated only after the events are inserted or updated to the db in which situation the context info of new value is lost.
So I add another value as a temporary store media

Reply:
oh I see. looks like needing at least two variables.